### PR TITLE
feat(TPG >=5.33)!: zone_distribution_config force recreation

### PR DESCRIPTION
**Pull request by lanandra**
Set default value for zone_distribution_config mode to MULTI_ZONE.

Refer to issue #220

### CODE REVIEW

**Comment by blueonrails** [comment]
Thanks for the PR, this bug is blocking us :+1:

File: modules/redis-cluster/main.tf [hunk 1]
```diff
@@ -32,6 +32,10 @@ resource "google_redis_cluster" "redis_cluster" {
     }
   }
 
+  zone_distribution_config {
```
Comment (imrannayer) [inline modules/redis-cluster/main.tf:35]: @lanandra can u plz make it dynamic block and also add `zone` to it?

Comment (anthony-delarosa-ts) [reply modules/redis-cluster/main.tf:35]: if I don't see any progress by end of week I'm gonna hijack this and create a new PR with suggested changes :D lol

File: modules/redis-cluster/main.tf [hunk 2]
```diff
@@ -32,6 +32,13 @@ resource "google_redis_cluster" "redis_cluster" {
     }
   }
 
+  dynamic "zone_distribution_config" {
+    for_each = var.zone_distribution_config_mode != "" ? [1] : []
+    content {
+      mode = var.zone_distribution_config_mode
+    }
+  }
+
```
Comment (lanandra) [inline modules/redis-cluster/main.tf:41]: @imrannayer `zone_distribution_config` has been updated to dynamic block, please review again. Thanks

Comment (imrannayer) [reply modules/redis-cluster/main.tf:41]: @lanandra `zone_distribution_config` has two properties `mode` and `zone`. Can you plz add `zone` to it?

File: modules/redis-cluster/main.tf [hunk 3]
```diff
@@ -32,6 +32,14 @@ resource "google_redis_cluster" "redis_cluster" {
...
   }
 
+  dynamic "zone_distribution_config" {
+    for_each = var.zone_distribution_config_mode != "" ? [1] : []
+    content {
+      mode = var.zone_distribution_config_mode
+      zone = var.zone_distribution_config_mode == "SINGLE_ZONE" ? var.zone_distribution_config_zone : null
+    }
+  }
+
```
Comment (lanandra) [inline modules/redis-cluster/main.tf:42]: @imrannayer updated to add zone inside `zone_distribution_config` dynamic block and only add the value inside zone if the mode is `SINGLE_ZONE` as referred to here: [[+](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/redis_cluster#nested_zone_distribution_config)]

Please review again. Thanks

Comment (imrannayer) [reply modules/redis-cluster/main.tf:42]: @lanandra Can you plz use null instead of empty string?

File: modules/redis-cluster/main.tf [hunk 4]
```diff
@@ -32,6 +32,14 @@ resource "google_redis_cluster" "redis_cluster" {
     }
   }
 
+  dynamic "zone_distribution_config" {
+    for_each = var.zone_distribution_config_mode != null ? [1] : []
```
Comment (lanandra) [inline modules/redis-cluster/main.tf:36]: @imrannayer empty string has been updated to null

File: modules/redis-cluster/versions.tf [hunk 5]
```diff
@@ -20,11 +20,11 @@ terraform {
 
     google = {
       source  = "hashicorp/google"
-      version = ">= 5.28, < 6"
+      version = ">= 5.33, < 6"
     }
     google-beta = {
       source  = "hashicorp/google-beta"
-      version = ">= 5.28, < 6"
+      version = ">= 5.33, < 6"
```
Comment (lanandra) [inline modules/redis-cluster/versions.tf:27]: provider versions has been updated to >= 5.33

**Review by imrannayer — APPROVED** [review]

**Pull request merged**